elastic/kibana
# [Form lib] Fix validation type execution when none is specified

I fixed a bug in the form lib that was running all validations types by default. This issue was found by @academo in one of the "cases" app form.

The form lib allows to specify different validation types. This is mainly used with the combobox field as in this case we need to be able to specify different validations for the **item** entered (e.g. it can not contain a certain char) and the **array** of items (e.g. the array can not be empty --> at least one item must be added).

The issue we found is that, by default, all the validations were executed on the field. Meaning that validations only meant to be ran on the array **item** (`type: VALIDATION_TYPES.ARRAY_ITEM`) were also executed on the array.

### Before

Notice when we remove the item from the combobox how the validation also runs and displays the "A value is required" error message

![before](https://user-images.githubusercontent.com/2854616/150142937-b5ac2470-dcda-40d8-9390-4fa23916ad62.gif)

### After

![after](https://user-images.githubusercontent.com/2854616/150143117-a44a360d-bcb2-4e8f-bbff-c2de3a1886ce.gif)

## How to test

* Modify the `x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/append.tsx` file and add `type: 'arrayItem'` on L41 (this marks this validation to only run on items being added to the combobox)
* Navigate to the ingest node pipeline ape, create a new pipeline and add a processor
* Select the "append" processor
* Try to add an empty item (multiple spaces) in the "Value" field
* Add now a valid item (e.g. "Foo")
* Remove the item from the combobox
* No error message should be displayed underneath

To test the previous behaviour revert the changes made to the `use_field.ts` file
